### PR TITLE
Add more types to Template class and compiled templates

### PR DESCRIPTION
**Pull request by ruudk**
Similar to https://github.com/twigphp/Twig/pull/4250.

### CODE REVIEW

**Comment by ruudk** [comment]
@fabpot @stof This is ready for review. Whenever you have time 😊

File: src/Template.php [hunk 1]
```diff
@@ -494,6 +490,7 @@ public function yieldParentBlock($name, array $context, array $blocks = [])
      *
      * @param array $context An array of parameters to pass to the template
      * @param array $blocks  An array of blocks to pass to the template
+     * @return iterable<string>
```
Comment (fabpot) [inline src/Template.php:494]: missing blank line above

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Template.php (hunk 1 above)
Comment (ruudk) [inline src/Template.php:494]:
```suggestion

     * @return iterable<string>
```

Comment (ruudk) [reply src/Template.php:494]: Done

**Comment by fabpot** [comment]
Thank you @ruudk.

**Pull request merged**